flow/src/main/java/com/sequenceiq/flow/service/FlowService.java

```diff
@@ -195,6 +201,29 @@ public FlowCheckResponse getFlowState(String flowId) {
         flowCheckResponse.setFlowId(flowId);
         flowCheckResponse.setHasActiveFlow(!completed("Flow", flowId, List.of(), allByFlowIdOrderByCreatedDesc));
         flowCheckResponse.setLatestFlowFinalizedAndFailed(isFlowInFailedState(allByFlowIdOrderByCreatedDesc, failHandledEvents));
+        // Get the latest flow information that will be used for error logging.
+        if (flowCheckResponse.getHasActiveFlow()) {
+            flowCheckResponse = getLatestKnownFlowCheckResponse(allByFlowIdOrderByCreatedDesc, flowCheckResponse, "flow", flowId);
+        }
+        return flowCheckResponse;
+    }
+
+    public FlowCheckResponse getLatestKnownFlowCheckResponse(List<FlowLogWithoutPayload> flowLogs, FlowCheckResponse flowCheckResponse, String type, String id) {
```

Comment:
Using the same `FlowCheckResponse` object as a return value AND a parameter too is a bad practice. You can change the return value to void. Please see the `setEndTimeOnFlowCheckResponse` method.

Reply:
It looks good, thank you!